src/main/java/com/xloop/resourceloop/authenticationservice/Controller/AuthController.java

```diff
@@ -76,7 +81,20 @@ public ResponseEntity<String> resetPasswordLink(@RequestBody String email){
         if(user == null){
             return ResponseEntity.status(403).body("Email Not Found");
         }
-        emailService.sendSimpleMessage(email, "Reset Password Email", "Hi, User....");
+        String otp = OTPGeneration.OTP(4);
+        ForgetPassword forgetPassword = new ForgetPassword(email,otp);
+        forgetPasswordRepo.save(forgetPassword);
+        emailService.sendSimpleMessage(email, "Reset Password Email", "Hi, User.... This is your otp "+otp);
         return ResponseEntity.status(200).body("Reset Pin Send Successfully");
     }
```

Comment:
@jeff-lent does this need to be more secure e.g:

- Expiry of OTPs 
- Cryptographically secure OTPs (we could send a link rather than ask them to input a code